tools/metams/metams_run_gc.xml

```diff
@@ -0,0 +1,347 @@
+                <param name="minclasssize" type="integer" value="3" label="min.class.size" help="The absolute number of samples in which a pseudospectrum is present before it is regarded as an unknown" />
+                <conditional name="options_db">
+                    <param name="option" type="select" label="Use Personnal DataBase option " >
+                        <option value="show">show</option>
+                        <option value="hide" selected="true">hide</option>
+                    </param>
+                    <when value="show">
+                        <param name="db_input" type="data" format="msp,txt" label=" DB file" help="A database file as needed for DB option in runGC" />
+                    </when>
+                    <when value="hide" />
+                </conditional>
+                <conditional name="options_ri">
+                    <param name="option" type="select" label="Use RI option " >
+                        <option value="show">show</option>
+                        <option value="hide" selected="true">hide</option>
+                    </param>
+                    <when value="show">
+                        <param name="ri_input" type="data" format="tabular,txt,csv,tsv,gg" label=" RI file" help="A file with two column (rt and RI) with rt in minutes as needed for RI option in runGC" />
+                    </when>
+                    <when value="hide" />
+                </conditional>
+            </when>
+        </conditional>
+        <conditional name="options_rifilter">
+            <param name="option" type="select" label="Use RI as filter" >
+                <option value="false" selected="true">FALSE</option>
+                <option value="true" >TRUE</option>
+            </param>
+            <when value="true">
+                <param name="ri_shift" type="integer" value="5" label="RIshift" help="RI shift accepted" />
+            </when>
+            <when value="false" />
+        </conditional>       
+        <!-- pour !is.null(DB) il faut un conditionnal avec un input type="text" pour charger la db au format msp et donc il faut définir un datatype (cf Misharl) -->
+    </inputs>
+
+    <outputs>
+
+        <data name="peaktable" format="tabular" from_work_dir="peaktable.tsv" label="peaktable.tsv" />
+        <data name="sampleMetadata" format="tabular" from_work_dir="sampleMetadata.tsv" label="sampleMetadata.tsv" />
+        <data name="variableMetadata" format="tabular" from_work_dir="variableMetadata.tsv" label="variableMetadata.tsv" />
+        <data name="dataMatrix" format="tabular" from_work_dir="dataMatrix.tsv" label="dataMatrix.tsv" />
+        <data name="peakspectra" format="txt" from_work_dir="peakspectra.msp" label="peakspectra.msp" />
+        <data name="rungcRData" format="rdata" from_work_dir="runGC.RData" label="runGC.RData" />
```

Comment:
It might be preferable to use `format="rds"`  for the output. In the R-script you might just switch to `saveRDS` and `readRDS`. But since you explicitly list the stored objects in the `save` call also `Rdata` might be acceptable. Some people have security concerns with `rdata`.

https://www.matheusmello.io/posts/r-what-are-the-main-differences-between-r-data-files